### PR TITLE
Provide the batch SQL analysis API

### PR DESCRIPTION
**Pull request by goldmedal**
# Description
Implement the batch SQL analysis API
## URL
```
GET /v2/analysis/sqls
```
## Request Body
- `manifestStr`: based64 manifset json string.
- `sqls`: the list of input sql.
```json
{
    "manifestStr": "ewogICJjYXRhbG9nIjogIndyZW5haSIsCiAgInNjaGVtYSI6ICJwdWJsaWMiLAogICJtb2RlbHMiOiBbCiAgICB7CiAgICAgICJuYW1lIjogIm9yZGVycyIsCiAgICAgICJyZWZTcWwiOiAic2VsZWN0ICogZnJvbSBvcmRlcnMiLAogICAgICAiY29sdW1ucyI6IFsKICAgICAgICB7CiAgICAgICAgICAibmFtZSI6ICJvcmRlcmtleSIsCiAgICAgICAgICAiZXhwcmVzc2lvbiI6ICJvX29yZGVya2V5IiwKICAgICAgICAgICJ0eXBlIjogImludGVnZXIiCiAgICAgICAgfSwKICAgICAgICB7CiAgICAgICAgICAibmFtZSI6ICJjdXN0a2V5IiwKICAgICAgICAgICJleHByZXNzaW9uIjogIm9fY3VzdGtleSIsCiAgICAgICAgICAidHlwZSI6ICJpbnRlZ2VyIgogICAgICAgIH0KICAgICAgXQogICAgfSwKICAgIHsKICAgICAgIm5hbWUiOiAiY3VzdG9tZXIiLAogICAgICAicmVmU3FsIjogInNlbGVjdCAqIGZyb20gY3VzdG9tZXIiLAogICAgICAiY29sdW1ucyI6IFsKICAgICAgICB7CiAgICAgICAgICAibmFtZSI6ICJjdXN0a2V5IiwKICAgICAgICAgICJleHByZXNzaW9uIjogImNfY3VzdGtleSIsCiAgICAgICAgICAidHlwZSI6ICJpbnRlZ2VyIgogICAgICAgIH0sCiAgICAgICAgewogICAgICAgICAgIm5hbWUiOiAibmFtZSIsCiAgICAgICAgICAiZXhwcmVzc2lvbiI6ICJjX25hbWUiLAogICAgICAgICAgInR5cGUiOiAidmFyY2hhciIKICAgICAgICB9CiAgICAgIF0KICAgIH0KICBdCn0=",
    "sqls": [
        "SELECT * FROM customer",
        "WITH t1 AS (SELECT * FROM customer) SELECT * FROM t1",
        "SELECT * FROM orders WHERE orderkey = 1 UNION SELECT * FROM orders where orderkey = 2"
    ]
}
```
The `manifestStr` is encoded from:
```json
{
  "catalog": "wrenai",
  "schema": "public",
  "models": [
    {
      "name": "orders",
      "refSql": "select * from orders",
      "columns": [
        {
          "name": "orderkey",
          "expression": "o_orderkey",
          "type": "integer"
        },
        {
          "name": "custkey",
          "expression": "o_custkey",
          "type": "integer"
        }
      ]
    },
    {
      "name": "customer",
      "refSql": "select * from customer",
      "columns": [
        {
          "name": "custkey",
          "expression": "c_custkey",
          "type": "integer"
        },
        {
          "name": "name",
          "expression": "c_name",
          "type": "varchar"
        }
      ]
    }
  ]
}
```

## Response
The result format is `List<List<QueryAnalysis>>`. You can check #548 for the detail of `QueryAnalysis`
```json
[
    // first sql result: SELECT * FROM customer
    [
        {
            "selectItems": [
                ...
            ],
            "relation": {
                ...
            },
            "groupByKeys": [],
            "sortings": [],
            "isSubqueryOrCte": false
        }
    ],
    // second sql result: WITH t1 AS (SELECT * FROM customer) SELECT * FROM t1
    [
        // for with query
        {
            "selectItems": [
                ...
            ],
            "relation": {
                ...
            },
            "groupByKeys": [],
            "sortings": [],
            "isSubqueryOrCte": true
        },
        // for the main query
        {
            "selectItems": [
                ...
            ],
            "relation": {
                ...
            },
            "groupByKeys": [],
            "sortings": [],
            "isSubqueryOrCte": false
        }
    ],
    // third sql result: SELECT * FROM customer UNION SELECT * FROM customer
    [
        // left of union
        {
            "selectItems": [
                ...
            ],
            "relation": {
                ...
            },
            "filter": {
                ...
            },
            "groupByKeys": [],
            "sortings": [],
            "isSubqueryOrCte": false
        },
        // right of union
        {
            "selectItems": [
                ...
            ],
            "relation": {
                ...
            },
            "filter": {
                ...
            },
            "groupByKeys": [],
            "sortings": [],
            "isSubqueryOrCte": false
        }
    ]
]
```

### CODE REVIEW

File: ibis-server/tests/routers/v2/test_analysis.py [hunk 1]
```diff
@@ -212,6 +212,25 @@ def test_analysis_sql_order_by():
...
+                "WITH t1 AS (SELECT * FROM customer) SELECT * FROM t1",
+                "SELECT * FROM orders WHERE orderkey = 1 UNION SELECT * FROM orders where orderkey = 2",
+            ],
+        }
+    )
+    assert len(result) == 4
+    assert len(result[0]) == 1
+    assert len(result[1]) == 1
+    assert len(result[2]) == 2
+    assert len(result[3]) == 2
```
Comment (grieve54706) [inline ibis-server/tests/routers/v2/test_analysis.py:231]: Could you add more expected results to let other contributors know the result format?

**Comment by grieve54706** [comment]
@goldmedal Could you add more information about the response content in the pull request? To explain why it needs two axes.

**Review by grieve54706 — APPROVED** [review]
LGTM, Thanks @goldmedal.

**Comment by goldmedal** [comment]
Thanks @grieve54706

**Pull request merged**